tools/java_api/src/main/java/com/kuzudb/KuzuStruct.java

```diff
@@ -0,0 +1,104 @@
+package com.kuzudb;
+
+public class KuzuStruct {
+    private Value structVal;
+
+    /**
+     * @return Gets the underlying Value for the struct
+     */
+    public Value getValue() {
+        return structVal;
+    }
+
+    /**
+     * Construct a struct from a value
+     *
+     * @param value the value to construct the struct from
+     */
+    public KuzuStruct(Value value) {
+        structVal = value;
+    }
+
+    /**
+     * Construct a struct literal from a given set of fields. The length of the
+     * fieldName/fieldValue arrays must be the same.
+     *
+     * @param fieldNames:  The name of the struct fields
+     * @param fieldValues: The values of the struct fields
+     */
+    public static KuzuStruct createStruct(String[] fieldNames, Value[] fieldValues) throws ObjectRefDestroyedException {
```

Comment:
Similarly this can be an constructor.

We can also add another one which constructs from a Java Map